### PR TITLE
Move normal() to DistributionTemplates

### PR DESCRIPTION
**Pull request by pbelevich**
The purpose of this PR is to move `normal`/`normal_`/`normal_out` to `native/DistributionTemplates.h`, `native/cpu/DistributionTemplates.h` and `native/cuda/DistributionTemplates.h` to make it reusable for custom RNG, see cpu_rng_test.cpp as an example of custom RNG.

Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#35167 Move normal() to DistributionTemplates**

Differential Revision: [D20588248](https://our.internmc.facebook.com/intern/diff/D20588248)

### CODE REVIEW

**Comment by pbelevich** [comment]
@ailzhang what can be cause XLA failures?
test_normal_xla_float64
test_randn_xla_float32
test_randn_xla_float64

**Comment by ailzhang** [comment]
cc: @JackCaoG who lowered `normal` :D. I assume we will not match `randn` result given we now have XLA-specific normal? 
Also we should likely fix `test_normal_xla_float64`.

File: aten/src/ATen/native/cuda/Copy.cu
Comment (anjali411) [inline aten/src/ATen/native/cuda/Copy.cu]: this is fixed in this PR https://github.com/pytorch/pytorch/pull/35344

**Review by anjali411 — APPROVED** [review]
lgtm after fixing the copy_() change :) thanks for the awesome work!

**Comment by pbelevich** [comment]
@ailzhang @JackCaoG should I wait for XLA change or land this PR?

File: aten/src/ATen/native/cuda/Copy.cu
Comment (pbelevich) [reply aten/src/ATen/native/cuda/Copy.cu]: removed

File: test/test_torch.py
Comment (ailzhang) [inline test/test_torch.py]: Curious: these two list are identical, is this intended?

Comment (pbelevich) [reply test/test_torch.py]: fixed, thanks!

**Comment by JackCaoG** [comment]
Sorry, missed the email earlier. I think you can land this pr, xla will skip new tests for now. I will investigate it a bit more, most likely test issue.

**Review by ailzhang — APPROVED** [review]
Thanks for waiting! This PR is good to go when you're ready. 
[note to self]:
It's fine to remove device dispatch for `normal` here since we have derivatives defined for 
```
normal(Tensor, double)
normal(double, Tensor)
normal(Tensor, Tensor)
```
So that they're still overridable by XLA. 
`normal(double, double)` is a composite op defined in `TensorFactories.cpp` which points to `normal_` as leaf node.

**Pull request closed without merging**
